enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/adm/ProjectVulnerability.java

```diff
@@ -162,12 +169,19 @@ public CompletableFuture<KnowledgeBaseItem> findKnowledgeBase(String knowledgeBa
                         .knowledgeBaseId(knowledgeBaseId).
                         build();
 
+                OCISessionInitiator session = OCIManager.getDefault().getActiveProfile();
+                String tenancyId = session.getTenancy().isPresent() ?
```

Comment:
A weird idea: Why not pass `OCISessionInitiator` to individual OCIItem & subclasses ? This logic (tenancy not present -> tenancyid/null + regioncode) could be done at one place (OCIItem). OCIItem can still have an alternative 'raw' constructor with IDs - this would be useful for TenancyItem.

Could it work?